refactor, net: Increase CNode data member encapsulation

hebasto opened the pull request:
All protected `CNode` data members could be private.

hebasto commented:
cc @troygiorshev @jnewbery

jnewbery commented:
Concept ACK making protected members private if they don't need to be protected. I'll review once #19673 is merged and this is rebased.

practicalswift commented:
Concept ACK

promag reviewed (commented):
Concept ACK.

I'd drop the "consolidate" commit. But if it's kept then it could be marked move-only. Also, there's an unnecessary `public:`.

sipa commented:
utACK 75f7e47b3ae59b7a8b1f5ec86ac281403a50a604

MarcoFalke commented:
Tend to NACK for the same reason: https://github.com/bitcoin/bitcoin/pull/19673#pullrequestreview-557045721

jnewbery commented:
@hebasto could you rebase now that #20816 is merged?

hebasto committed:
refactor, move-only: Relocate CNode private members

hebasto committed:
refactor, net: Increase CNode data member encapsulation

All protected CNode data members could be private.

hebasto commented:
Updated 75f7e47b3ae59b7a8b1f5ec86ac281403a50a604 -> 7f20edf4bb827a368b8eaf8d2c22643d5935c613 ([pr20373.01](https://github.com/hebasto/bitcoin/commits/pr20373.01) -> [pr20373.02](https://github.com/hebasto/bitcoin/commits/pr20373.02)).

@jnewbery 
> @hebasto could you rebase now that #20816 is merged?

Done.

@promag 
> ... if it's kept then it could be marked move-only.

Done.

> Also, there's an unnecessary `public:`.

Fixed.

jnewbery commented:
ACK 7f20edf4bb827a368b8eaf8d2c22643d5935c613

Verified that this is move-only, except for making the protected members private.

Thanks for cleaning this up!

hebasto commented:
Updated 7f20edf4bb827a368b8eaf8d2c22643d5935c613 -> 3642b2ed34e6609e8de558b352516daadb12cac1 ([pr20373.02](https://github.com/hebasto/bitcoin/commits/pr20373.02) -> [pr20373.03](https://github.com/hebasto/bitcoin/commits/pr20373.03)) due to the conflict with #20881.

jnewbery commented:
utACK 3642b2ed34e6609e8de558b352516daadb12cac1

MarcoFalke commented:
review ACK 3642b2ed34e6609e8de558b352516daadb12cac1 🏛

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

review ACK 3642b2ed34e6609e8de558b352516daadb12cac1 🏛
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUgpSgwAiwjdT13xFBxwJQmARjPmO8p8A7XcH2TDGrYMjehHgN9z/c6LajnsGLrf
HxU581vUH1HnWmjnSFdlQOKC/IDRiUDw9ogpakbjVi4Z1+KJnXEwJCDoCQepAKGA
adpBXg1jZLbLXlyhsiladQb5mAEZ0Cbru8NnXIriUP0SFgM7RGR7EaP9G0BUVqYx
3rpDxXtsUmtaZ7Y2/2SpTABh2UEK0ya1v1x8ILlWwz1YhqTEunSI8VwNffxCiVBn
V+oYGbKEyM/zg6x3MQnnX1plXgoz0XEOqmqqK12jtkTQAknWp7tQj7DhSCbSjLaS
ojE2LPFc5GhYDA0nvMq8N//Bm6yd6Z29WK5x8trAGmOEbpGxcyqrjN4Nnuw0O0Ai
hxt0XnQ6YMjeFvh5xzrIQEQwTd2bttrk6CiAf46hnwwmXwHbNwecTRQtj4+7LZlf
frOLEwBnFnQiOZjkiIXP0Q1L3pfQYKSNxr8b426QUBhQBNLojP+xBw4eLX3Gu3Iq
zSgvcVPd
=leU8
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `a0e7b38da83edfebd9fb41f063e4ec58f924d3f47816589b67b81a023d3dcd94  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108a0e7b38da83edfebd9fb41f063e4ec58f924d3f47816589b67b81a023d3dcd94f0102ea7dff9179c279e65e3b38fa6f0461f08fff0102082af81e86eb80af9942ac6f9c4643c08f1045ffc3fd1f008a9aeba754343da9e0083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff0106e5654fefb05afb1a66c5ebb171b271408f1045ffc3fd1f00882a033d9479f4c620083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6df010975badea419118b25c591d8ad3d9e15008f1045ffc3fd1f008defe94ea4cee01830083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6d)

</details>
